### ISSUE TITLE
Automatically deploy master to staging/dev server

### ISSUE THREAD

**Issue by s4wny** [sjchat/sjchat#42]
Automatically deploy master to staging/dev server.

Setup git hooks / docker hooks.

**Comment by s4wny** [comment]
Plan

Set up a webhook server that can build our docker files and push them to docker hub.
Is this basically what jenkins does?

```
Master branch is updated
	-> github webhook notify our webhook server
		-> Build our docker files
		-> Pushe them to our docker hub with tag :latest

When someone pushes to the docker hub.
	-> docker hub call notify our webhook server
		-> Servern SSH into our staging machine and run `docker-machine stack deploy`
```

**Comment by s4wny** [comment]
How to SSH into the docker-machines created by docker-machine create:

https://github.com/docker/machine/issues/3212

**Comment by Krarl** [comment]
[Travis documentation for docker hub](https://sebest.github.io/post/using-travis-ci-to-build-docker-images/)

**Comment by s4wny** [comment]
We have now implemented a travis after_success script that uploads our docker images to docker hub.

We could create a new service, sjchat-auto-deployment which would create a webserver with the url `/updateServer&password=sjflkjsafjs` and then call the URL with a webhook from docker hub.

The `/updateServer` server should then:
1. Download the latest `docker-compose-staging.yaml` file from this repo.
2. Run `sudo docker stack deploy -c docker-compose-staging.yaml sjchat`, which would update the server by downloading the latest docker images from docker hub.

The `&password=` thing is to prevent unathorized persons from triggering the task.


This auto deployment should only deploy to staging I guess. Depolyment to production is still something we can do by hand.

**Comment by Krarl** [comment]
Resolved by #78
